### PR TITLE
rust ci

### LINKED ISSUE #9: Add a GitHub Action
It needs to:
- Build the project
- Run the tests
- Check formatting (i.e. `cargo fmt`)
- Run linter (i.e. `cargo clippy -D warnings`)

### PR DESCRIPTION
**Pull request by ts0yu**
closes https://github.com/lambdaclass/libtorrent-rs/issues/9

### CODE REVIEW

File: .github/workflows/rust.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+on:
+  push:
+    branches:
+      - master
```
Comment (Oppen) [inline .github/workflows/rust.yaml]: There's no `master` branch.

File: .github/workflows/rust.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+      - uses: Swatinem/rust-cache@v1
+        with:
+          cache-on-failure: true
+
+      - name: cargo fmt
+        run: cargo +nightly fmt --all -- --check
+
+      - name: cargo clippy
+        run: cargo +nightly clippy --all --all-features -- -D warnings
```
Comment (Oppen) [inline .github/workflows/rust.yaml]: Add the missing newline.

**Review by Oppen — CHANGES_REQUESTED** [review]

File: .github/workflows/rust.yaml (hunk 1 above)
Comment (ts0yu) [reply .github/workflows/rust.yaml]: fixed

File: .github/workflows/rust.yaml (hunk 2 above)
Comment (ts0yu) [reply .github/workflows/rust.yaml]: what do you mean by this?

Comment (unbalancedparentheses) [reply .github/workflows/rust.yaml]: i believe you were missing an eol and gits needs it. we should add it to the options of the repo actually to add it automatically.

**Comment by unbalancedparentheses** [comment]
thanks for this! the team will check it and merge it.

**Comment by Oppen** [comment]
Quick question, why are the changed targets using nightly?

**Comment by ts0yu** [comment]
it's just for the lints - I believe there are some on nightly that aren't on stable. this is just the default ci template I use for most rust projects

**Comment by Oppen** [comment]
I would recommend against it. Using nightly for the lints could randomly break the workflows.

**Comment by ts0yu** [comment]
I'll push changes soon.

**Comment by ts0yu** [comment]
removed nightly

**Review by Oppen — APPROVED** [review]

**Review by juanbono — APPROVED** [review]

**Pull request merged**